kii_thing_if.c

```diff
@@ -684,6 +686,30 @@ static kii_bool_t prv_onboard_with_vendor_thing_id(
             return KII_FALSE;
         }
     }
+    if (firmware_version != NULL) {
+        if (kii_api_call_append_body(kii, ",\"firmwareVersion\":",
+                        CONST_STRLEN(",\"firmwareVersion\":")) != 0) {
+            M_KII_LOG(kii->kii_core.logger_cb("request size overflowed.\n"));
+            return KII_FALSE;
+        }
+        if (kii_api_call_append_body(kii, firmware_version,
```

Comment:
@rizakii 
This is good:+1: but I found another issuer reported by @yongpingchen san.

I often forget to add `"` I also want to prevent this. So I want to fix this like following with your suggestion and yonping san's suggestion:

```c
  if (append_key_value("vendorThingID", vendorThingID, NOT_SUCCESSOR) != 0
          append_key_value_optional("layoutPosition", layoutPosition,
                  IS_SUCCESSOR) != 0) {
      return KII_FALSE
  }
```

What do you think?